service/service.go

```diff
@@ -20,22 +20,26 @@ const (
 	UNKNOWN   = iota
 )
 
+type Namer interface {
```

Comment:
Oh, I see you made an interface. But I think it will be cleaner to totally separate the implementations and just pass them a service to name.